source/core/WorldBase.hpp

```diff
@@ -101,6 +102,44 @@ namespace cse491 {
       return *agent_set.back();
     }
 
+    /// @brief Build a new entity
+    /// @tparam PROPERTY_Ts Types for any properties to set at creation (automatic)
+    /// @param entity_name The name of this agent
+    /// @param properties Name/value pairs for any properties set at creation
+    /// @return A reference to the newly created entity
+    template <typename... PROPERTY_Ts>
+    Entity & AddEntity(std::string entity_name="None", PROPERTY_Ts... properties) {
+        auto entity_ptr = std::make_unique<Entity>(item_set.size(), entity_name);
+        entity_ptr->SetProperties(std::forward<PROPERTY_Ts>(properties)...);
+        item_set.emplace_back(std::move(entity_ptr));
+        return *item_set.back();
+    }
+
+    /// @brief Remove an agent from the agent set
+    /// @param agent_name The name of this agent
+    /// @return None
+    void RemoveAgent(std::string agent_name="None") {
+        if (agent_name == "Interface")
```

Comment:
What is the goal of having an Interface just `return` without being removed or taking any other action?  One option might be to mark the run as over, but that would cause problems for multi player games.  Probably the best option is to remove the interface like you would any other agent and trust that the interface's destructor will take care of everything.  A better option might be to have a function that is run to notify an agent or impending removal and then marking them as inactive.  This is a topic we should talk about as a class.